src/hist/basehist.py

```diff
@@ -327,27 +317,29 @@ def plot2d(
         if self.ndim != 2:
             raise TypeError("Only 2D-histogram has plot2d")
 
-        """
-        Default Figure: construct the figure and axes
-        """
-        if fig is None:
-            fig = plt.figure(figsize=(8, 8))
-            grid = fig.add_gridspec(4, 4, hspace=0, wspace=0)
+        # Default Figure: construct the figure and axes
+        if fig is not None and ax is not None:
```

Comment:
Please take a look at this fig/ax handling, the other functions need it too. If you pass fig and ax, the ax must be in the figure. If you pass one, the other is created or accessed.